### DISCUSSION TITLE
ZMQ pubhash?

### DISCUSSION THREAD

**Discussion by MiningCryptoLive** [dreams-money/merged-mining-pool#14] [Q&A]
I setup the pool and the pool starts and after a minute it stops with the error  panic: zmq4: could not dial to "tcp://localhost:1222" connect: connection refused. I added the port to the firewall group but still does not work. Any help is appreciated

**Comment by dreams-money** [comment]
Do you have your ZMQ configured? - https://github.com/dreams-money/merged-mining-pool?tab=readme-ov-file#getting-started

**Reply by MiningCryptoLive** [reply]
yes both Dogecoin and Litecoin nodes have zmqpubhashblock="tcp://0.0.0.0:1222" in the conf files and nodes restarted. If I am unable to get that to work will the pool run without it?

**Reply by MiningCryptoLive** [reply]
Now I have a new issue. panic: zmq4: could not open a ZMTP connection: zmq4: could not initiate ZMTP connection: zmq4: could not exchange greetings.

Man this stuff is hard!! LOL

**Reply by BenzMo** [reply]
Hey, not a dev here but sounds like a ZMQ connection issue, I would make sure any relevant dependencies are installed and test the connection outside the pool software first (e.g. with a script) to narrow down the issue. 

Also since you're using 0.0.0.0 which I believe is specified for remote host usage in the getting started file. Is your mining pool config file pointing to the right remote server ip address? I only mention as your first message included a localhost attempted dial, which if it's on a remote host doesn't sound right to me.

**Reply by MiningCryptoLive** [reply]
> Hey, not a dev here but sounds like a ZMQ connection issue, I would make sure any relevant dependencies are installed and test the connection outside the pool software first (e.g. with a script) to narrow down the issue.
> 
> Also since you're using 0.0.0.0 which I believe is specified for remote host usage in the getting started file. Is your mining pool config file pointing to the right remote server ip address? I only mention as your first message included a localhost attempted dial, which if it's on a remote host doesn't sound right to me.

I changed it to 127.0.0.1 because the pool is on the same server. Now I get the error I posted. Not sure where to go from here. I telnet 127.0.0.1 port and it connects so the port is open. I will keep fiddling to see if I can find the issue.

**Reply by BenzMo** [reply]
These are my config settings within the Litecoin and Dogecoin nodes:

server=1
rpcallowip=127.0.0.1
rpcport=22555
rpcuser=dogeuser
rpcpassword=dogepass
zmqpubrawblock=tcp://127.0.0.1:1220
zmqpubrawtx=tcp://127.0.0.1:1221
zmqpubhashblock=tcp://127.0.0.1:1222

server=1
rpcallowip=127.0.0.1
rpcport=9332
rpcuser=liteuser
rpcpassword=litepass
zmqpubrawblock=tcp://127.0.0.1:1225
zmqpubrawtx=tcp://127.0.0.1:1224
zmqpubhashblock=tcp://127.0.0.1:1223

My mining pool server config file uses the `zmqpubhashblock` one as the specified connection variable. Hope that helps. Obviously user/pass not correct and should be changed.

**Reply by MiningCryptoLive** [reply]
Thanks so much... I changed my conf files and restarted the nodes. waited a couple minutes for them to catch up and now it all works!!!!!!!!!

**Reply by MiningCryptoLive** [reply]
Now to work on a WEB-UI.

**Reply by BenzMo** [reply]
All good 😄 glad you got it working

**Reply by MiningCryptoLive** [reply]
So after finally getting the pool to run now I can't connect to it with L7. Get socket read error on the pool side.

**Reply by BenzMo** [reply]
Try setting the timeout in the config file to 600s (what I have it at). Edit: deleted the miner software bit as I think I found success with Cgminer initially but pretty sure Godminer and other software also work for me now.

**Reply by MiningCryptoLive** [reply]
Using Godminer. I will change the timeout and see if that works. Thanks

**Reply by BenzMo** [reply]
I'd also be patient with it (a few minutes ish to be safe), I think sometimes for me it has taken a while to connect and get the first valid share.

**Reply by MiningCryptoLive** [reply]
Changing the timeout did it. Rig is connected and submitting valid shares. Thanks for all your help on this!

**Reply by BenzMo** [reply]
Hooray! All good

**Reply by MiningCryptoLive** [reply]
![image](https://github.com/user-attachments/assets/a309ca1a-16db-4522-a628-b678fbb5cfea)
